CMakeLists.txt

```diff
@@ -82,7 +82,7 @@ option(DISABLE_RCT2 "WIP: Try building without using code and data segments from
 option(USE_MMAP "Use mmap to try loading rct2's data segment into memory.")
 option(LAUNCHPAD_BUILD "Sets some build system options on launchpad.net")
 
-set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations")
+set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Warray-bounds -Waddress -Wchar-subscripts -Wenum-compare")
```

Comment:
@Manuel-K I've just hit that on my system. Further investigation showed this warning is only really present with `-O3` (aka `CMAKE_BUILD_TYPE=release`), which I never build.

You can work your way around it by going to `-O2` or `-Os` for this file manually (`make VERBOSE=1` will show you exact command, simply replace appropriate option and run `make` again, it should pick your object file) or just do `RelWithDebInfo`/`MinSizeRel`.